src/constructLayoutEngine.js

```diff
@@ -198,7 +199,7 @@ function insertNode(node, container, previousSibling) {
   if (previousSibling && previousSibling.nextSibling) {
     // move to be immediately after previousSibling
     previousSibling.parentNode.insertBefore(node, previousSibling.nextSibling);
-  } else if (node.parentNode !== container) {
+  } else {
```

Comment:
This fixes a bug - we cannot assume the the previous location of the node within its parent container (before the route change) will be its new position within the parent.